### DISCUSSION TITLE
How can i use spline agent to analysize downloaded spark execution plan file?

### DISCUSSION THREAD

**Discussion by liujiawen** [AbsaOSS/spline#1201] [Q&A] [answered]
Hi, my production environment does not allow me to add a listener to the Spark shells. Therefore, I get the execution plan from the Spark history server API. Is it possible for Spline agent to read the downloaded execution plan instead of a listener?

Thanks.

**Comment by wajda** [comment, marked as answer]
Answered in https://github.com/AbsaOSS/spline-spark-agent/issues/659.
It is potentially possible, but that would be another Spline agent project that would interact with or integrate into a Spark History server, collect the lineage data and send it to the Spline server over the Spline Producer API.
